entware_install.sh

```diff
@@ -48,6 +53,45 @@ else
     fi
 fi
 
+if [ ! -d /home/root/.cache/wget_bin ]
+then
+    # Bootstrap a current version of wget
+    WGET_BINARIES_PATH='http://static.cosmos-ink.net/remarkable/artifacts'
+    WGET_BINARIES_FILENAME='wget-remarkable-pipeline_job245_wget1.20.3.zip'
+    WGET_BINARIES_SHA256='84185a5934e34e25794d439c78dc9f1590e4df12fbf369236f6a8749bf14d67f'
+
+    # Download and compare to hash
+    wget "$WGET_BINARIES_PATH/$WGET_BINARIES_FILENAME" -O "/home/root/$WGET_BINARIES_FILENAME"
```

Comment:
That's a good point. I currently host the file on my Raspberry Pi 4 at home which has a decent internet connection and 99.9% uptime (see [here for more](https://github.com/toltec-dev/toltec/pull/38#discussion_r493058710)). I could also easily switch the domain to a VPS I also have. Or we can just upload the artifact to github or somewhere else and link to there (though I'm not sure whether github allowed and http connection or older tls for https).

Reply:
The two options are to retrieve a wget binary which supports TLS over http, or to install entware over http and install another wget through entware at the end.

I don't think either case is any less secure than the other, so I'd just prefer the simpler case of installing wget at the end after entware is installed (over http).

Reply:
The idea of installing wget before entware, was that I would only have one unsecure downlod (which is verified by the included hash though). Even though I didn't change all URLs, yet most parts of the installation can then also be made using proper https before entware is fully installed and thus greatly reducing insecure downloads over http.

Reply:
Personally I'd prefer pulling down wget first so that the rest of the network traffic can be over https. Otherwise we'd want some mechanism to validate all the files that entware pulls down until we can flip it over to using https, which is less than ideal.

Reply:
@Eeems But if the initial wget binary is compromised, you have the same issue.

Reply:
I created [an issue](https://github.com/Entware/Entware/issues/537) in the entware project.  We'll see what they say, but for the moment, let's not do the extra wget stuff.

Reply:
The only potential insecurity would be the user downloading this script with wget. If a user scps `entware_install.sh` to his/her reMarkable, there is no problem. The script has the hash and the wget binary is part of the downloaded zip file.

I agree though that the point where the artifacts is hosted is not the most reliable. We should reupload the artifact somewhere, where it will always be available. @Eeems pointed out that raw.githubusercontent.com supports using http. So that should works as well.

Reply:
> 
> 
> @Eeems But if the initial wget binary is compromised, you have the same issue. In its current state, the proposed PR doesn't add much security. You'd need to include a hash of the wget binary with the install script.
> 
> I don't see much difference between downloading 1 file unsecurely from one server (the wget binary), vs downloading a handful of files unsecurely from another server (entware). Also, I expect entware's repo to still be online 5 years from now, but not necessarily a third party server.

I didn't realize we didn't have the hash of the binary included. I'd suggest we should add hash checking.

> 
> 
> @Eeems pointed out that raw.githubusercontent.com supports using http. So that should works as well.
It technically doesn't, but the built-in wget won't fail pulling using https

Reply:
> I didn't realize we didn't have the hash of the binary included. I'd suggest we should add hash checking.

We do. The zip containing the wget binary is checked. That's basically as good as hash checking the wget binary directly.


> I don't see much difference between downloading 1 file unsecurely from one server (the wget binary), vs downloading a handful of files unsecurely from another server (entware).

@Evidlo I don't think you mean't https://github.com/toltec-dev/toltec/pull/38. But this one really needs hash checking as well. Sadly that will defeat the purpose of a one-liner. Maybe we get a 2 or 3 liner with that. Still better then defeating the whole point of the security at that entry-point.  
This is a matter of that issue though. Maybe we should make this issue wait until that's sorted out there. It was already mentioned in the discord and will very likely become a topic there soon as well.